ed_sensor_integration/src/laser/laser_plugin.cpp

```diff
@@ -21,240 +22,46 @@
 
 #include <tue/profiling/timer.h>
 
-namespace
-{
-
-typedef std::vector<unsigned int> ScanSegment;
 
-struct EntityUpdate
+/**
+ * copy a ros laserscan message to a standard vector
+ *
+ * @param[in] scan laserscan message
+ * @param[out] sensor_ranges vector with ranges
+ */
+void lasermsgToSensorRanges(const sensor_msgs::LaserScan::ConstPtr& scan, std::vector<float>& sensor_ranges)
```

Comment:
I think we should able to do less copying by choosing the correct data structure(s)